sklearn/decomposition/sparse_pca.py

```diff
@@ -98,7 +98,7 @@ def __init__(self, n_components=None, alpha=1, ridge_alpha=0.01,
         self.verbose = verbose
         self.random_state = random_state
 
-    def fit(self, X, y=None):
+    def fit(self, X, y=None, variance=False):
```

Comment:
The `fit` method should not have any other param than X and y (and sometimes sample_weight). You have to make it a class attribute I guess.

Reply:
ok sounds good, moved to `__init__`